pkg/ingester/instance.go

```diff
@@ -213,32 +226,68 @@ func (i *instance) Label(_ context.Context, req *logproto.LabelRequest) (*logpro
 	}, nil
 }
 
-func (i *instance) lookupStreams(req *logproto.QueryRequest, matchers []*labels.Matcher, filter logql.Filter) ([]iter.EntryIterator, error) {
+func (i *instance) Series(_ context.Context, req *logproto.SeriesRequest) (*logproto.SeriesResponse, error) {
+	groups, err := loghttp.Match(req.GetGroups())
+	if err != nil {
+		return nil, err
+	}
+
+	dedupedSeries := make(map[uint64]logproto.SeriesIdentifier)
+	for _, matchers := range groups {
+		err = i.forMatchingStreams(matchers, func(stream *stream) error {
+			// exit early when this stream was added by an earlier group
+			key := stream.labels.Hash()
+			if _, found := dedupedSeries[key]; found {
+				return nil
+			}
+
+			dedupedSeries[key] = logproto.SeriesIdentifier{
+				Labels: stream.labels.Map(),
+			}
+			return nil
+		})
+
+		if err != nil {
+			return nil, err
+		}
+	}
+	series := make([]logproto.SeriesIdentifier, 0, len(dedupedSeries))
+	for _, v := range dedupedSeries {
+		series = append(series, v)
+
+	}
```

Comment:
Rather than a for loop here, can we do one of the following?

```
series = append(series, dedupedSeries...) 
```

or 

```
copy(series, dedupedSeries)
```

Reply:
`dedupedSeries` is a `map[uint64]logproto.SeriesIdentifier` while `series` is `[]logproto.SeriesIdentifier` so I need to loop across the deduplicated keys.

Reply:
Oh, duh, sorry 🙃